Docs: Clarify PHP Usage

lukeapage opened the pull request:
Clarify that PHP is not required for testing and add a link to the CONTRIBUTING page.

I was a bit confused by the Readme and it seems to be out of date. Sorry if I am wrong about the tests not requiring PHP, but they seem to run without it and its alot less hassle to not need a PHP server.

scottgonzalez commented:
I forget if we ever actually required PHP for unit tests, but I believe you're correct that we don't require them for any tests now.

I still feel pretty strongly that a first time contributor with a small patch (which is a decent portion of our contributors) shouldn't need to install node. Perhaps we should have both minimal requirements and recommended setups documented. Ideally you run `grunt` to do full linting and testing, but many contributors can get away with just running unit tests in their browsers.

lukeapage commented:
I think that would make things clearer - I can change to that.

re: the contributors section, I thought that was a bit unclear - there are *loads* of links and the ones in the readme (I didn't notice CONTRIBUTING at first) point to the wiki page - where some of the links e.g. Commit Message Style Guide go to invalid pages.

Perhaps it is better someone experienced re-writes or looks at those paragraphs - but I could certainly do that with some guidance - otherwise I'll just clean up the testing requirements.

lukeapage commented:
Okay, I've shuffled things around a bit, I think it is a further improvement.

scottgonzalez commented:
> re: the contributors section, I thought that was a bit unclear - there are loads of links and the ones in the readme (I didn't notice CONTRIBUTING at first) point to the wiki page - where some of the links e.g. Commit Message Style Guide go to invalid pages.

Thanks for pointing that out, I'll go through and clean that up. We've done a lot of work to try to consolidate this information on the new contribute.jquery.org site. I guess there are still a few older pages that have links that haven't been updated to point to the new content.

scottgonzalez commented on CONTRIBUTING.md:
```diff
@@ -18,15 +18,20 @@ Every week (unless otherwise noted) the jQuery UI team has a meeting to discuss
-* [Node.js](http://nodejs.org/) (includes NPM, necessary for the next step)
-* Grunt (install with: `npm install -g grunt`)
+* Get [Node.js](http://nodejs.org/) (includes NPM, necessary for the next step)
+* Install Grunt cli (install with: `npm install -g grunt`)
```
`npm install -g grunt-cli`

scottgonzalez commented on README.md:
```diff
@@ -21,16 +21,18 @@ In general, fork the project, create a branch for a specific change and send a
 - [Mongoose (most platforms)](http://code.google.com/p/mongoose/)
+- [http-server](https://www.npmjs.com/package/http-server)
 
+You can also run the unit tests inside phantomjs by [installing node](http://nodejs.org/) and the required modules(`npm install`) and then running `grunt test`.
```
Let's replace "installing node and the required modules" with "setting up your environment" and linking to CONTRIBUTING.md#environment-minimum-required.

scottgonzalez replied:
We may want to do something similar int he Building jQuery UI section on line 41 (with a note that the web server isn't needed).

scottgonzalez commented:
Overall, this looks good to me. I'd like to wait for some others to chime in though.

lukeapage commented:
I've adjusted based on comments and also made a few further change - `Build a Local Copy of jQuery UI` in contributing actually does not tell you how to build a local copy of jquery - it contained some information on how to fork and some information on how to run the tests. I've moved those into their own sections. Also made the title casings consistent and improved the duplication/organisation of information.

I understand this is a little controversial and that everyone will have their own opinion on the best wording, but I am sure this is better than it is currently, so will just see if anyone else has any feedback.

scottgonzalez commented on CONTRIBUTING.md:
```diff
@@ -16,68 +16,83 @@ This is the best way to contribute to jQuery UI. Please read through the full gu
+### Environment: Minimum Required
 
-jQuery UI uses Node.js & Grunt to automate the building and validation of source code.
+If you are contributing changes you will need a fork of jquery (see [Getting the Source](#environment-getting-the-source)). If you just want the source code you could clone jquery ui directly:
```
a fork of jquery-ui

scottgonzalez replied:
"clone jquery ui directly"

It's always either jquery-ui (the repo) or jQuery UI (the project).

scottgonzalez commented on CONTRIBUTING.md:
```diff
@@ -16,68 +16,83 @@ This is the best way to contribute to jQuery UI. Please read through the full gu
+### Environment: Getting the Source
 
-Change directory to your web root directory, whatever that might be:
+* Create a fork of the jQuery UI repo on GitHub at http://github.com/jquery/jquery-ui. This will create a fork of jquery-ui at your username.
```
"at your username" -> "in your GitHub account"

scottgonzalez commented:
Thanks for the updates. I left comments for a few small tweaks, but this looks great.

scottgonzalez commented on README.md:
```diff
@@ -21,35 +21,21 @@ In general, fork the project, create a branch for a specific change and send a
 
 ## Building jQuery UI
 
 jQuery UI uses the [Grunt](http://github.com/gruntjs/grunt) build system.
```
While we're making changes, we should probably update this to link to http://gruntjs.com/ instead.

lukeapage commented:
tweaks done

scottgonzalez commented:
Thanks.
